public/app/features/dashboard-scene/variables/utils.ts

```diff
@@ -26,3 +30,219 @@ export const createDependencyEdges = (variables: Array<SceneVariable<SceneVariab
 
   return edges;
 };
+
+export interface VariableUsageTree {
+  variable: SceneVariable<SceneVariableState>;
+  tree: unknown;
+}
+
+export interface UsagesToNetwork {
+  variable: SceneVariable<SceneVariableState>;
+  nodes: GraphNode[];
+  edges: GraphEdge[];
+  showGraph: boolean;
+}
+
+export function createUsagesNetwork(variables: Array<SceneVariable<SceneVariableState>>, dashboard: Dashboard) {
```

Comment:
I've copied these utils from https://github.com/grafana/grafana/blob/396e75dadf8646c42e4d7f509bc38d1ade5afe83/public/app/features/variables/inspect/utils.ts and adjusted them to work with SceneVariable. In that file there are some other utils that are not used in dashboards, and the utils I copied are specific to dashboards so I think it makes sense having them here vs updating the functions to support SceneVariable - especially since we are planning on deprecating the old arch at some point

Reply:
Yes! This is the right approach! 🙌